In [1]:
import pandas as pd

df=pd.read_csv('./Quality_cost.csv')

#### Making analysis ####

In [3]:
df.head()

Day  Weeks  Shifts  Worked (hour) Day to produce cassette  \
0     Monday     52       2              8                     Yes   
1    Tuesday     52       2              8                     Yes   
2  Wednesday     52       2              8                      No   
3   Thursday     52       2              8                     Yes   
4     Friday     52       2              8                      No   

   Poor quality in cassette  Raw material  Average cassette OoS  
0                      5670         509.6                    23  
1                      5670         509.6                    23  
2                      5670         509.6                    23  
3                      5670         509.6                    23  
4                      5670         509.6                    23

# Making Poor Quality Cost Analysis 

A review of the MEDISA company identified the costs associated with poor quality, which will only be considered in relation to non-compliance with specifications in the production of Polystyrene case.

MEDISA works 2 shifts of 8 hours each from Monday to Friday and 2 shifts of 4 hours each on Saturdays. 

Mondays, Tuesdays and Thursdays are set aside for the Polystyrene case manufacturing process.

## Number of incidents occurring in a period of time


### On average, there are 23 polystyrene cases per  day that do not comply with specifications

In [33]:
incidents_per_day = 23
polystyrene_days_worked_per_week = 3

### Determinig the labor costs associated with the repair (rework)

In [34]:
# Calculating associates
pre_expansion = 1
molding = 1
cutting = 2
shipping = 0
people_workin_in_area = pre_expansion + molding + cutting + shipping
# Calculating work hours
#worked_hours_per_day_weekdays = 8
#shift_weekdays = 2
#hours_worked_weekdays = worked_hours_per_day_weekdays * shift_weekdays
# Labor cost per day
labor_cost_per_day = 5670
# (incidents per day) * (number of people that work in the area) * (worked hours per day) * (hourly rate of pay)

labor_cost = incidents_per_day * people_workin_in_area * labor_cost_per_day

labor_cost_money = "${:.2f}".format(labor_cost)
labor_cost_money

'$521640.00'

#### Cost of rework = $521,640.00

### Determining the material costs associated with defects

In [35]:
raw_material_costs = 509.6
material_costs_with_defects = incidents_per_day * raw_material_costs

material_costs_with_defects_money = "${:.2f}".format(material_costs_with_defects)
material_costs_with_defects_money

'$11720.80'

#### Cost of material associated with defects = $11,720.80

# Cost Poor Quality result - day

In [36]:
cost_poor_quality_day = labor_cost + material_costs_with_defects

cost_poor_quality_day_money = "${:.2f}".format(cost_poor_quality_day)
cost_poor_quality_day_money

'$533360.80'

#### Cost of poor Quality per day = $533,360.80

In [38]:
cost_poor_quality_year = cost_poor_quality_day * polystyrene_days_worked_per_week * 52

cost_poor_quality_year_money = "${:.2f}".format(cost_poor_quality_year)
cost_poor_quality_year_money

'$83204284.80'

#### Cost of poor Quality per year = $83,204,284.80

# Beneficts of apply Lean Six Sigma project

In [42]:
# The objective is to reduce at 80% of failures in manufacturing process of polystyrene cases
beneficts = "${:.2f}".format(cost_poor_quality_year * 0.2)
beneficts

'$16640856.96'

#### Cost of poor Quality per year = $16,640,856.96
#### A savings of $16,640,856.96 per year will be realized by reducing variations in the process